In [1]:
%run thebigone.ipynb
%run flag_complex.ipynb

Miguel recommended doing this for the forcing term; I'll have to track this down and find it.  

In [2]:
data = Gaussian_ts()

In [3]:
data.shape


(3, 10000)

In [15]:
truncdata = data[:,-10000:]
truncdata.shape

(6, 10000)

In [5]:
def howmanyvertices(ts):
    for i in range(1,20):
        seq = bin_sol(ts,i)
        A, verts = adjacency(seq)
        print(f"for {i} bins there are {len(verts)} vertices and {np.count_nonzero(A)} edges")

In [6]:
howmanyvertices(data)

for 1 bins there are 1 vertices and 1 edges
for 2 bins there are 8 vertices and 64 edges
for 3 bins there are 27 vertices and 350 edges
for 4 bins there are 50 vertices and 763 edges
for 5 bins there are 88 vertices and 1426 edges
for 6 bins there are 134 vertices and 2423 edges
for 7 bins there are 186 vertices and 3650 edges
for 8 bins there are 254 vertices and 4866 edges
for 9 bins there are 322 vertices and 6132 edges
for 10 bins there are 416 vertices and 7207 edges
for 11 bins there are 518 vertices and 8054 edges
for 12 bins there are 627 vertices and 8663 edges
for 13 bins there are 754 vertices and 9117 edges
for 14 bins there are 876 vertices and 9412 edges
for 15 bins there are 1003 vertices and 9553 edges
for 16 bins there are 1154 vertices and 9716 edges
for 17 bins there are 1311 vertices and 9804 edges
for 18 bins there are 1462 vertices and 9830 edges
for 19 bins there are 1640 vertices and 9871 edges


In [8]:
def tstoph(ts,numbins):
    seq = bin_sol(ts,numbins)
    G = construct_digraph(seq)
    D = distance_matrix(G)
    dist = D.reshape(1,*D.shape)
    ph = flagser_persistence.fit_transform(dist)
    return ph

In [15]:
barcode = tstoph(data,9)

Stats for the last 10000 points of L63_sampled (3D)

|Bins| Vertices | Edges | runtime (ts to flagser barcode in dim 0 and 1) |
|---|-----|-----|---|
|6 |75|207|3.2s|
|7 | 102|290| 3.8s|
|11|226|609|32.7s|
|15|389|1007|7m44.6s|
|18| 533|1357|56m51.5s|

Stats for all 1250000 points of L_63 sampled (3D)
|Bins|Vertices|Edges|runtime|
|---|---|---|---|
|7|107|403|32.8s|
|10|217|828|50.2s|
|11|261|970|2m25.4s|
|12|306|1139|3m1.7s|
|14|419|1573|11m19.5s|
|16|542|2038|67m6.1s|


One interesting thing about this is that it makes it seem like the computational bottleneck is the number of vertices and not the number of edges. The lower bins take longer for this one because how how huge the timeseries is. 

Stats for the L63 function generated by the diffeqs - 50000 points (3D)

|Bins| Vertices | Edges | runtime (ts to flagser barcode in dim 0 and 1) |
|---|-----|-----|---|
|8|113|294|8.8s|
|11|203|508|30.5s|
|14|304|743|3m11.4s|
|19|516|1242|59m41.4s|


L96 - all 100000 points. (8D)

The dimensionality is kind of nuts. Definitely need to reduce that. 

|Bins|Vertices|Edges|runtime|
|---|---|---|---|
|2|209|1132|43.8s|
|3|982|3819| killed my work laptop lol|

Truncated L96 - last 10000 points, I don't think this one should have been dead.

|Bins|Vertices|Edges|runtime|
|---|---|---|---|
|2|148|471|41.2s|
|3|494|1282| also dead |

CDV - this one is six dimensional so I'll be curious to see how different it works than the 8 dimensional

This will be for the full 2000000 points

|Bins| Vertices | Edges | runtime (ts to flagser barcode in dim 0 and 1) |
|---|-----|-----|---|
|3 |89|409|40.5s|
|5 |374|1750|5m30.4s|


Trunc CDV - the number of points played a huge role in how long it takes to build the graphs, the dimension seems not to matter too much, but does matter hugely in the persistence

|Bins| Vertices | Edges | runtime (ts to flagser barcode in dim 0 and 1) |
|---|-----|-----|---|
|6|376|1065|12m8.3s|
|8|607|1496|more than 30|

JetLet is 3 dimensional with only 9810 vertices.

|Bins| Vertices | Edges | runtime (ts to flagser barcode in dim 0 and 1) |
|---|-----|-----|---|
|6|110|1074|17.0s|
|8|206|2017|34.3s|
|10|327|3113|5m24.5s|
|12|505| 4596|36m25.4s|

The Gaussian is 3D with 10000 points

|Bins|Vertices|Edges|runtime(ts to flagser barcode in dim 0 and 1)|
|---|---|---|---|
|4|50|763|3.6s|
|5|88|1426|5.1s|
|6|134|2423|9.4s|
|7|186|3650|19.2s|
|8|254|4866|57.2s|
|9|322|6132|2m46.8s|


for 9 bins there are 322 vertices and 6132 edges
for 10 bins there are 416 vertices and 7207 edges
for 11 bins there are 518 vertices and 8054 edges
for 12 bins there are 627 vertices and 8663 edges
for 13 bins there are 754 vertices and 9117 edges
for 14 bins there are 876 vertices and 9412 edges
for 15 bins there are 1003 vertices and 9553 edges
for 16 bins there are 1154 vertices and 9716 edges
for 17 bins there are 1311 vertices and 9804 edges
for 18 bins there are 1462 vertices and 9830 edges
for 19 bins there are 1640 vertices and 9871 edges